include/uv.h

```diff
@@ -1096,6 +1104,9 @@ struct uv_fs_s {
   void* ptr;
   const char* path;
   uv_stat_t statbuf;  /* Stores the result of uv_fs_stat() and uv_fs_fstat(). */
+  const uv_dir_t* dir; /* Stores the result of uv_fs_opendir() */
+  uv_dirent_t* dirents;
+  size_t nentries;
```

Comment:
And where would you put those members or the new struct? FYI, I'll be -1 on bending over backwards for ABI compatibility in new features. That way we'd never start working on v2.